airbyte-cdk/java/airbyte-cdk/core/src/main/kotlin/io/airbyte/cdk/core/IntegrationCommand.kt

```diff
@@ -0,0 +1,165 @@
+import io.airbyte.cdk.integrations.base.JavaBaseConstants
+import io.airbyte.cdk.integrations.util.ApmTraceUtils
+import io.airbyte.cdk.integrations.util.ConnectorExceptionUtil
+import io.airbyte.protocol.models.v0.AirbyteConnectionStatus
+import io.airbyte.protocol.models.v0.AirbyteMessage
+import io.github.oshai.kotlinlogging.KotlinLogging
+import io.micronaut.context.annotation.Value
+import jakarta.inject.Inject
+import jakarta.inject.Named
+import java.nio.file.Path
+import java.util.Optional
+import java.util.function.Consumer
+import picocli.CommandLine
+
+private val logger = KotlinLogging.logger {}
+
+/** CLI implementation that invokes the requested operation for the connector. */
+@CommandLine.Command(
+    name = "airbyte-connector",
+    description = ["Executes an Airbyte connector"],
+    mixinStandardHelpOptions = true,
+    header =
+        [
+            "@|magenta     ___    _      __          __       |@",
+            "@|magenta    /   |  (_)____/ /_  __  __/ /____   |@",
+            "@|magenta   / /| | / / ___/ __ \\/ / / / __/ _   |@",
+            "@|magenta  / ___ |/ / /  / /_/ / /_/ / /_/  __/  |@",
+            "@|magenta /_/  |_/_/_/  /_.___/\\__, /\\__/\\___/|@",
+            "@|magenta                    /____/              |@",
+        ],
+)
+@SuppressFBWarnings(
+    value = ["NP_NONNULL_RETURN_VIOLATION", "NP_OPTIONAL_RETURN_NULL"],
+    justification = "Uses dependency injection",
+)
+class IntegrationCommand : Runnable {
+    @Value("\${micronaut.application.name}") lateinit var connectorName: String
+
+    @Inject lateinit var operations: List<Operation>
+
+    @Inject
+    @Named("outputRecordCollector")
+    lateinit var outputRecordCollector: Consumer<AirbyteMessage>
+
+    @CommandLine.Parameters(
+        index = "0",
+        description =
+            [
+                "The command to execute (check|discover|read|spec|write)",
+                "\t check - checks the config can be used to connect",
+                "\t discover - outputs a catalog describing the source's catalog",
+                "\t read - reads the source and outputs messages to STDOUT",
+                "\t spec - outputs the json configuration specification",
+                "\t write - writes messages from STDIN to the integration",
+            ],
+    )
+    lateinit var command: String
+
+    @CommandLine.Option(
+        names = ["--" + JavaBaseConstants.ARGS_CONFIG_KEY],
+        description =
+            [
+                JavaBaseConstants.ARGS_CONFIG_DESC,
+                "Required by the following commands: check, discover, read, write",
+            ],
+    )
+    lateinit var configFile: Path
+
+    @CommandLine.Option(
+        names = ["--" + JavaBaseConstants.ARGS_CATALOG_KEY],
+        description =
+            [
+                JavaBaseConstants.ARGS_CATALOG_DESC,
+                "Required by the following commands: read, write",
+            ],
+    )
+    lateinit var catalogFile: Path
```

Comment:
Shouldn't this be of type `Path?` or `Optional<Path>`?

Reply:
@postamar It doesn't have to be because its not actually being used to load the value.  It's mainly here so that the usage will print out the required argument.  I will add a comment indicating this AND change it to `Optional` just to be consistent.

Reply:
I'm cool with whatever you do as long as it's not surprising :)

Reply:
I personally prefer `Path?`, with the default value being set to null, as it would be if the option isn't specified

Reply:
@stephane-airbyte The reason for `Optional` here is because that is how PicoCLI handles arguments that are not required.  Generally speaking, we would use `Path?`, but in this case we need to use Optional to conform to the framework.